### ISSUE TITLE
[BUG] ForecastingHorizon does not work for periods of N time units, N>1, e.g., 30min `PeriodIndex`

### ISSUE THREAD

**Issue by manujosephv** [alan-turing-institute/sktime#1435]
**Describe the bug**
Although the example in the documentation works, AutoETS and ExponentialSmoothing errors out when we use a sub-daily frequency of data

Error Message:
```
KeyError                                  Traceback (most recent call last)
<ipython-input-49-df19e8e04e62> in <module>()
      1 forecaster = AutoETS(sp=12, auto=True,n_jobs=-1)
      2 forecaster.fit(y)
----> 3 y_pred = forecaster.predict(fh=[1,2,3])

6 frames
/usr/local/lib/python3.7/dist-packages/pandas/core/indexing.py in _validate_read_indexer(self, key, indexer, axis, raise_missing)
   1296             if missing == len(indexer):
   1297                 axis_name = self.obj._get_axis_name(axis)
-> 1298                 raise KeyError(f"None of [{key}] are in the [{axis_name}]")
   1299 
   1300             # We (temporarily) allow for some missing keys with .loc, except in

KeyError: "None of [PeriodIndex(['2021-09-20 01:00', '2021-09-20 01:30', '2021-09-20 02:00'], dtype='period[30T]', freq='30T')] are in the [index]"
```
**To Reproduce**

```python
import numpy as np
import pandas as pd
from sktime.forecasting.ets import AutoETS

freq = "30T"
_y = np.arange(50) + np.random.rand(50) + np.sin(np.arange(50)/4)*10
t = pd.date_range("2021-09-19", periods=50, freq=freq)
y = pd.Series(_y, index=t)
y.index = y.index.to_period(freq=freq)
forecaster = AutoETS(sp=12, auto=True,n_jobs=-1)
forecaster.fit(y)
y_pred = forecaster.predict(fh=[1,2,3])
```

**Expected behavior**

Forecast for the future without an error

**Additional context**

**Versions**
<details>

System:
    python: 3.7.12 (default, Sep 10 2021, 00:21:48)  [GCC 7.5.0]
executable: /usr/bin/python3
   machine: Linux-5.4.104+-x86_64-with-Ubuntu-18.04-bionic

Python dependencies:
          pip: 21.1.3
   setuptools: 57.4.0
      sklearn: 0.24.2
       sktime: 0.8.0
  statsmodels: 0.12.2
        numpy: 1.19.5
        scipy: 1.7.1
       Cython: 0.29.24
       pandas: 1.1.5
   matplotlib: 3.4.3
       joblib: 1.0.1
        numba: 0.54.0
     pmdarima: 1.8.2
      tsfresh: 0.18.0

</details>

**Comment by mloning** [comment]
Thanks for the bug report @manujosephv! 

The error happens in the `to_absolute_int` method of the `ForecastingHorizon` object when called inside `_predict` of the forecaster. 

The issue is that even though we subtract a period from a period index, with both having the same frequency, we do not get the difference as the number of periods, but as minutes. The returned object is an index consisting of `pandas._libs.tslibs.offsets.Minute` objects.

**Comment by fkiraly** [comment]
This should be fixed by #1981.
